panic: runtime error: index out of range [0] with length 0 (v5.98.0_x5)

### Terraform and AWS Provider Version

```console
Terraform v1.11.0
on linux_amd64
Your version of Terraform is out of date! The latest version
is 1.12.1. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s) or Data Source(s)

* `aws_security_group_rule`
* `aws_elasticache_replication_group`
* `aws_elasticache_cluster`
* `aws_cloudfront_distribution`
* `aws_kinesis_firehose_delivery_stream`
* `aws_wafv2_web_acl`

### Expected Behavior

should work since we didn't change anything on these resources

### Actual Behavior

errors with the same reason

### Relevant Error/Panic Output

e.g.

```sh
│ Error: Plugin did not respond
│ 
│   with <redacted>,
│   on .<redacted> line <redacted>, in resource "aws_elasticache_replication_group" "cache-cluster":
│   <redacted>: resource "aws_elasticache_replication_group" "cache-cluster" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
```

```sh
Stack trace from the terraform-provider-aws_v5.98.0_x5 plugin:
panic: runtime error: index out of range [0] with length 0
goroutine 27511 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/elasticache.resourceReplicationGroupRead({0x1df5f928, 0xc007c52e10}, 0xc005340b00, {0x1b7ba9a0?, 0xc003400c00?})
	github.com/hashicorp/terraform-provider-aws/internal/service/elasticache/replication_group.go:721 +0x1ed9
github.com/hashicorp/terraform-provider-aws/internal/provider.wrapResource.(*wrappedResource).read.interceptedHandler[...].func4(0xc005340b00, {0x1b7ba9a0, 0xc003400c00})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:114 +0x3b6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1df5f928?, {0x1df5f928?, 0xc005d2f440?}, 0xd?, {0x1b7ba9a0?, 0xc003400c00?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:850 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001129e00, {0x1df5f928, 0xc005d2f440}, 0xc00accc1a0, {0x1b7ba9a0, 0xc003400c00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:1149 +0x525
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc003362ff0, {0x1df5f928?, 0xc005d2ed80?}, 0xc006477680)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:739 +0x6a5
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xc000df7c80, {0x1df5f928?, 0xc005d2e9c0?}, 0xc006477680)
	github.com/hashicorp/terraform-plugin-mux@v0.18.0/tf5muxserver/mux_server_ReadResource.go:35 +0x196
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00150f900, {0x1df5f928?, 0xc005d2e0f0?}, 0xc007aff730)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:784 +0x2f6
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x1b6032a0, 0xc00150f900}, {0x1df5f928, 0xc005d2e0f0}, 0xc005340680, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:575 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc003358600, {0x1df5f928, 0xc005d2e060}, 0xc005ed8c60, 0xc00349ba10, 0x2a1b5628, 0x0)
	google.golang.org/grpc@v1.72.0/server.go:1405 +0x103b
google.golang.org/grpc.(*Server).handleStream(0xc003358600, {0x1df97140, 0xc003654000}, 0xc005ed8c60)
	google.golang.org/grpc@v1.72.0/server.go:1815 +0xbaa
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.72.0/server.go:1035 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 104
	google.golang.org/grpc@v1.72.0/server.go:1046 +0x125
Error: The terraform-provider-aws_v5.98.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Sample Terraform Configuration

most of params are variables

<details>
<summary>Click to expand configuration</summary>

```hcl
resource "aws_elasticache_replication_group" "cache-cluster" {
  cluster_mode               = var.cluster_mode
  replication_group_id       = var.cache_name
  description                = "${var.env}-${var.name} cluster"
  node_type                  = var.redis_node_type
  auto_minor_version_upgrade = false
  port                       = 6379
  engine                     = var.engine
  engine_version             = var.engine_version
  parameter_group_name       = var.parameter_group_name
  automatic_failover_enabled = true
  multi_az_enabled           = var.multi_az_enabled
  // availability_zones            = data.aws_ec2_instance_type_offerings.zones.locations
  preferred_cache_cluster_azs = length(var.availability_zones) > 0 ? var.availability_zones : null

  subnet_group_name    = var.subnet_group_name
  security_group_ids   = concat(var.security_group_ids, [aws_security_group.cache_security_group.id])
  security_group_names = var.security_group_names

  apply_immediately         = var.apply_immediately
  maintenance_window        = var.maintenance_window
  snapshot_window           = var.snapshot_window
  snapshot_retention_limit  = var.snapshot_retention_limit
  final_snapshot_identifier = local.final_snapshot_identifier

  replicas_per_node_group = var.replica_count
  num_node_groups         = var.shard_count

  tags = merge(
    local.default_tags,
    var.tags,
  )

  lifecycle {
    ignore_changes = [replication_group_id]
  }
}
```
</details>


### Steps to Reproduce

1. create a resource and a state(i guess under 5.98 would be work)
2. plan again with the provider with the version 5.98

### Debug Logging

<details>
<summary>Click to expand log output</summary>

```console

```
</details>


### GenAI / LLM Assisted Development

n/a

### Important Facts and References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/42726.txt
+++ b/.changelog/42726.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_elasticache_replication_group: Fix crash during read operations where configuration endpoint and node groups are nil and empty, respectively
+```

--- a/internal/service/elasticache/replication_group.go
+++ b/internal/service/elasticache/replication_group.go
@@ -715,7 +715,7 @@ func resourceReplicationGroupRead(ctx context.Context, d *schema.ResourceData, m
 	if rgp.ConfigurationEndpoint != nil {
 		d.Set(names.AttrPort, rgp.ConfigurationEndpoint.Port)
 		d.Set("configuration_endpoint_address", rgp.ConfigurationEndpoint.Address)
-	} else {
+	} else if len(rgp.NodeGroups) > 0 {
 		log.Printf("[DEBUG] ElastiCache Replication Group (%s) Configuration Endpoint is nil", d.Id())
 
 		if rgp.NodeGroups[0].PrimaryEndpoint != nil {

